logic/source_map.go

```diff
@@ -22,7 +22,12 @@ func DecodeSourceMap(ism map[string]interface{}) (SourceMap, error) {
 	sm := SourceMap{}
```

Comment:
Alternative:
```
	buf:=json.Encode(ism)
	json.Decode(buf,&sm)
```

Reply:
Can you help me understand why this is better?

Reply:
The only reason I bring it up, is that it's easy to get things wrong with reflection. So if there's a way to avoid using it the code would be easier to reason about and less likely that there is an error. So basically, I'm asking if these are simpler alternatives.

> Why are you unmarshaling to a map[string]interface{} instead of a structured object?

What I mean is you could use Unmarshal into a real type instead of `map[string]interface{}`, that way `Unmarshal` would handle all of this logic. Looking a bit closer just now, I wonder if this is even better:
```
func DecodeSourceMap(ism map[string]interface{}) (SourceMap, error) {
  buf := json.Encode(ism)
  var sourcemap SourceMap
  return json.Decode(buf, sourcemap)
}
```

Reply:
I can do that to begin with to set the fields specified but I still need to do some post-processing to decode the VLQ values. Is it worth it?

Reply:
I think it is, but not enough to block your PR so I'll leave it up to you.

Reply:
I'm going to merge this and post a follow up with the above since I think this is tripping folks up right now